### DISCUSSION TITLE
re: Oldplay opk and RG280V issues

### DISCUSSION THREAD

**Discussion by hongho71** [eduardofilo/RG350_adam_image#241] [Q&A]
Anybody get Oldplay opk and RG280V with the latest Adam Image to work?  Whenever I boot Oldplay up and try to play mp3 files, it boots me out.  What is weird is when I try to play video mp4 file, it actually plays the sound portion fine just not mp3 files.  When I use RG280V_ROGUE firmware, Oldplay plays mp3 fine but can't seem to play mp3 files on the Adam Image.

**Comment by neilswann80** [comment]
I hadn't actually tried Oldplay as GMU (incorporated within Adam Image - apps section) works very well for playing MP3s.  I've just tested Oldplay on my RG350M and am getting the exact same situation you've described.

**Comment by hongho71** [comment]
Forgot to mention GMU on Anbernic RG280V Adam Image plays mp3 fine but the whole screen is tiny small font while it looks normal on the RG280V_Rogue firmware.  In a pinch, I could use GMU but can hardly see the screen or just swap out the microsd card for RG280V_Rogue firmware.  But hopefully Adam Image can get an update to fix GMU and/or Oldplay opk apps to play mp3 without the bugs.  Thanks.

**Reply by neilswann80** [reply]
Yeah, GMU is definitely like the bottom row of an optician's vision test chart!!!

I've tried playing with the config file inside the OPK:
>SDL_frontend.Width=320
>SDL_frontend.Height=240

But no dice!  Frame Buffer related?

**Reply by eduardofilo** [reply]
@plrguez has fixed the resolution problem of GMU in RG280. The new OPK is now included in the repository (commit de93635735141bcff84f1c2185389c5353b27f85) but until a new image release appears you can install it manually by replacing the copy in `/media/data/apps`.

https://github.com/plrguez/gmu-odbeta/releases/tag/v0.10.2-rc1-odbeta-20220521

**Reply by a-dekker** [reply]
For those who want to fix the extremely tiny font-size on the RG350m:
Add the following line to `~/.config/gmu/gmu.conf`
`SDL.DefaultSkin=default-modern-large`

**Reply by eduardofilo** [reply]
Great tip. I will try to add that configuration to the image if I find a good compromise that works well at the different resolutions. Thanks @a-dekker .

**Reply by eduardofilo** [reply]
There is no good compromise. The `default-modern-large` skin looks too big on RG280 and `default-modern` too small on RG350M. What a pity.

**Reply by a-dekker** [reply]
Indeed. A solution would be to auto-select the skin based on the screen resolution. I think a small change in the code could do that. We could open an issue in the gmu-odbeta repo or try to create a PR for that ourselves?
Edit: I noticed that creating issues is disabled in the repo.

**Reply by a-dekker** [reply]
And I have done a brave attempt for a PR: https://github.com/plrguez/gmu-odbeta/pull/1

**Reply by plrguez** [reply]
Sorry for late reply. The code in the PR would not solve the problem.

- The opk uses a shell script to launch gmu (https://github.com/plrguez/gmu-odbeta/blob/sdl12/gmu-gcw0.dge) that copy the default config file (https://github.com/plrguez/gmu-odbeta/blob/sdl12/gmu.rg350.conf) as gmu.conf in $HOME/.config/gmu/ path if not exists
- This default config file have the option SDL.DefaultSkin=default-modern
- The code in PR is using cfg_add_key_if_not_present function, that will not add SDL.DefaultSkin value as it is already present in the config

But we have various options:

- Add a specific configuration file for RG350M and adapt the shell to install it. This requires remove $HOME/.config/gmu/gmu.conf if exists to take effect
- Use the source code in a-dekker PR and remove the SDL.DefaultSkin option in the default file gmu.rg350.conf. Again this requires remove $HOME/.config/gmu/gmu.conf if exists
- Change the source code in the PR to use the cfg_add_key function instead of cfg_add_key_if_not_present. But with this option, the user could never change this option with the configuration file.

I prefer the first option. No changes to base code and the user have the possibility to change the option to their taste editing the config file in $HOME/.config/gmu/gmu.conf

**Reply by a-dekker** [reply]
Hmmm, you are right, I thought the option SDL_DefaultSkin was not set in the config file by default, but that's not the case unfortunately.
All 3 option you mention seem valid to me. Acting based on actual resolution instead a devicename is a bit more generic, but we do not have that many models.

If you choose the first option I'll cancel my PR. I could create a new PR based on option 1, but I think you can implement that yourself with ease.
Thanks in advance!

**Reply by plrguez** [reply]
I have implemented the first option and pushed the changes to the repo. I have tested on RG280M and RG350M and seems to work. 

The new defaults only be applied if `$HOME/.config/gmu/gmu.conf`does not exists. So it must be deleted if you want it.

Any feedback is welcome.

The new release can be downloaed [here](https://github.com/plrguez/gmu-odbeta/releases/tag/v0.10.2-rc1-odbeta-20220526)

**Reply by a-dekker** [reply]
Looks good. I removed my config files and after starting GMU I get the proper font on my RG350M.

**Reply by eduardofilo** [reply]
Tested and works perfect. Thank you very much @plrguez . I have already incorporated this release to the image.

**Reply by Coccijoe** [reply]
Hi, just a question, are you able to seekforward and backward in the GMU version include in Adam?  For me only the timecode change. When i use L1+R2 the sound continue to play without seek.

**Reply by eduardofilo** [reply]
True, the same thing happens to me. I guess that given the delicate engineering done to adapt the GMU code to the new audio libraries of the system, not all the functions of the program work.
